src/messaging/ReliableMessageManager.cpp

```diff
@@ -0,0 +1,654 @@
+        {
+            // Paustime is specified in milliseconds; Update retrans values
+            RetransTable[i].nextRetransTimeTick =
+                static_cast<uint16_t>(RetransTable[i].nextRetransTimeTick + (PauseTimeMillis >> mTimerIntervalShift));
+        } // exchContext
+    }     // for loop in table entry
+
+    // Schedule next physical wakeup
+    StartTimer();
+}
+
+/**
+ * Return a tick counter value given a time period.
+ *
+ * @param[in]  newTime        Timestamp value of in milliseconds.
+ *
+ * @return Tick count for the time period.
+ */
+uint64_t ReliableMessageManager::GetTickCounterFromTimePeriod(uint64_t period)
+{
+    return (period >> mTimerIntervalShift);
+}
+
+/**
+ * Return a tick counter value between the given time and the stored time.
+ *
+ * @param[in]  newTime        Timestamp value of in milliseconds.
+ *
+ * @return Tick count of the difference between the given time and the stored time.
+ */
+uint64_t ReliableMessageManager::GetTickCounterFromTimeDelta(uint64_t newTime)
+{
+    return GetTickCounterFromTimePeriod(newTime - mTimeStampBase);
+}
+
+#if defined(RMP_TICKLESS_DEBUG)
+void ReliableMessageManager::TicklessDebugDumpRetransTable(const char * log)
+{
+    ChipLogProgress(ExchangeManager, log);
+
+    for (int i = 0; i < CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE; i++)
+    {
+        if (RetransTable[i].rc)
+        {
+            ChipLogProgress(ExchangeManager, "EC:%04" PRIX16 " MsgId:%08" PRIX32 " NextRetransTimeCtr:%04" PRIX16,
+                            RetransTable[i].rc, RetransTable[i].msgId, RetransTable[i].nextRetransTimeTick);
+        }
+    }
+}
+#else
+void ReliableMessageManager::TicklessDebugDumpRetransTable(const char * log)
+{
+    return;
+}
+#endif // RMP_TICKLESS_DEBUG
+
+/**
+ * Iterate through active exchange contexts and retrans table entries.  If an
+ * action needs to be triggered by ReliableMessageProtocol time facilities,
+ * execute that action.
+ */
+void ReliableMessageManager::ExecuteActions()
+{
+#if defined(RMP_TICKLESS_DEBUG)
+    ChipLogProgress(ExchangeManager, "ReliableMessageManager::ExecuteActions");
+#endif
+
+    ExecuteForAllContext([](ReliableMessageContext * rc) {
+        if (rc->IsAckPending())
+        {
+            if (0 == rc->mNextAckTimeTick)
+            {
+#if defined(RMP_TICKLESS_DEBUG)
+                ChipLogProgress(ExchangeManager, "ReliableMessageManager::ExecuteActions sending ACK");
+#endif
+                // Send the Ack in a Common::Null message
+                rc->SendCommonNullMessage();
+                rc->SetAckPending(false);
+            }
+        }
+    });
+
+    TicklessDebugDumpRetransTable("ReliableMessageManager::ExecuteActions Dumping RetransTable entries before processing");
+
+    // Retransmit / cancel anything in the retrans table whose retrans timeout
+    // has expired
+    for (int i = 0; i < CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE; i++)
+    {
+        ReliableMessageContext * rc = RetransTable[i].rc;
+        if (rc)
```

Comment:
how about inverting the conditions to decrease the indent level? Something like:

```
for (int i = 0; i < CHIP_CONFIG_RMP_RETRANS_TABLE_SIZE; i++)
{
￼   ReliableMessageContext * rc = RetransTable[i].rc;
￼   if (!rc) {
       continue;
    }

    if (0 == RetransTable[i].nextRetransTimeTick) {
       continue;
    }

    .....
```

Reply:
done